x-pack/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -169,8 +170,17 @@ export const createCustomThresholdExecutor = ({
     const hasGroups = !isEqual(groups, [UNGROUPED_FACTORY_KEY]);
     let scheduledActionsCount = 0;
 
+    const alertLimit = baseAlertFactory.alertLimit.getValue();
+
     // The key of `groups` is the alert instance ID.
     for (const group of groups) {
+      if (scheduledActionsCount >= alertLimit) {
+        // need to set this so that warning is displayed in the UI and in the logs
+        baseAlertFactory.alertLimit.setLimitReached(true);
```

Comment:
This is only conditionally reported back. We should report back `false` if we've successfully sent all the alerts, so I I guess after the for loop

Reply:
The rule execution fails if the number of alerts is less than alert limit. Also, I noticed that in this case, alert is generated, but with every rule execution, new alert is created instead of updating status of current alert.

<img width="1258" alt="Screenshot 2023-12-01 at 14 08 25" src="https://github.com/elastic/kibana/assets/69037875/d9b049d8-c9d4-4066-b40b-99446c125ab7">

<img width="650" alt="Screenshot 2023-12-01 at 14 10 16" src="https://github.com/elastic/kibana/assets/69037875/f7c0273a-6324-453a-82b0-1e3c8a078bb0">

Reply:
@ymao1 so we break the loop after setting         
`baseAlertFactory.alertLimit.setLimitReached(true);
`
so we need to set it to false immediately after that?

Reply:
<img width="1465" alt="image" src="https://github.com/elastic/kibana/assets/3505601/69267c1f-dbb4-4a4c-89bf-b78702745384">
now it seems to be working , after i set limit false